datastore/memcached.go

```diff
@@ -0,0 +1,103 @@
+package datastore
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/bradfitz/gomemcache/memcache"
+	"github.com/flashbots/mev-boost-relay/common"
+)
+
+const (
+	defaultMemcachedExpirySeconds = 60
+)
+
+var (
+	ErrInvalidProposerPublicKey = errors.New("invalid proposer public key specified")
+	ErrInvalidBlockHash         = errors.New("invalid block hash specified")
+)
+
+type Memcached struct {
+	client    *memcache.Client
+	keyPrefix string
+}
+
+// SaveExecutionPayload attempts to insert execution engine payload to memcached using composite key of slot,
+// proposer public key, block hash, and cache prefix if specified. Note that writes to the same key value
+// (i.e. same slot, proposer public key, and block hash) will overwrite the existing entry.
+func (m *Memcached) SaveExecutionPayload(slot uint64, proposerPubKey, blockHash string, payload *common.GetPayloadResponse) error {
+	if m == nil {
```

Comment:
not a huge fan of this pattern, nil check should be responsibility of the caller? otherwise it lends itself to magic/unclear behavior 🤔

Reply:
discussed offline, resolved